### ISSUE TITLE
Handle Ref/Promise migration

### ISSUE THREAD

**Issue by SystemFw** [functional-streams-for-scala/fs2#1135]
We are in the process of moving Ref and Promise to cats-effect, which means we'd need to update fs2 accordingly. I'll be doing this work since I've got some spare time this week.

**Comment by SystemFw** [comment]
https://github.com/typelevel/cats-effect/pull/198

**Comment by mpilquist** [comment]
Okay these are unblocked now -- `Ref`, `Deferred`, and `Semaphore` are in the latest cats-effect.

**Comment by SystemFw** [comment]
The changes are looking a bit bigger and a lot less mechanical than I had anticipated, but it's coming along nicely. Should be done by the end of the weekend (`core` done, `io` and `test` to go)

**Comment by SystemFw** [comment]
Fixed in #1137
